adr/2025_03_10-suggestions/diff-yjs-snapshots.md

```diff
@@ -0,0 +1,40 @@
+
+# Diff two Y.js snapshots
+
+Y.js keeps track of metadata around what changes have been made to a document in order to efficiently synchronize changes. So, to compare a document with it's previous version, you can leverage this metadata by taking a snapshot of the full history of a document, and compare the insertions and deletions that have been made between the two snapshots. Because they share history, you can figure out what has been modified from _this point_ in history compared to _that point_ in history. For clarity, let's call this a `y-diff` for now to make sure it is different than a full content diff.
+
+## Representing these changes in the editor
+
+I won't go into detail about how to implement the `y-diff` between two documents, given that it is partially implemented in y-prosemirror, but will instead focus on how those changes end up being represented within the editor.
+
+We have a few different options:
+
+### Changes as _attributes_
+
+Kevin has already implemented a preliminary version of the `y-diff` between two documents. Specifically, by using `ychange` attributes to add metadata onto elements to indicate whether they were modified between the two versions. This approach works for simple documents at the moment, with some limitations around:
+
+- Certain features are missing (due to this being preliminary work) like
+  - XYZ
```

Comment:
I would be super interested in feedback here. I remember that the approach - albeit buggy in certain scenarios - is solid.